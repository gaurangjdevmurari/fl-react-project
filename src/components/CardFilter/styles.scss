.CardFilter {
  display: flex;
  flex-direction: row-reverse;
  margin-right: 20px;

  .sort-by {
    align-content: end;
  }
}
