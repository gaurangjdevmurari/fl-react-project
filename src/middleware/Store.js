import React, {useReducer} from 'react';
import {initialState, reducer} from "./reducers/FilterReducer";

export const FilterStore = React.createContext({});

function Store({children}) {
    let [movieState, movieDispatch] = useReducer(reducer, initialState);
    return (
        <FilterStore.Provider value={[movieState, movieDispatch]}>
            {children}
        </FilterStore.Provider>
    );
}

export default Store;
