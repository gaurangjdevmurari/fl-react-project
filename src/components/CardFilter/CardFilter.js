import React from 'react';
import "./styles.scss";
import {Button, Select} from "antd";
import {Option} from "antd/lib/mentions";

function CardFilter(props) {
    return (
        <div className="CardFilter">
            <div className="sort-by">
                Sort By: {" "}
                <Select defaultValue="lucy" style={{width: 120}} onChange={() => {

                }}>
                    <Option value="jack">Jack</Option>
                    <Option value="lucy">Lucy</Option>
                    <Option value="disabled" disabled>
                        Disabled
                    </Option>
                    <Option value="Yiminghe">yiminghe</Option>
                </Select>
            </div>
            <Button type="link">Show All Classes</Button>
        </div>
    );
}

export default CardFilter;
