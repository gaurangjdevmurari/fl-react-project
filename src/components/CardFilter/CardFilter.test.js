import React from 'react';
import {createRenderer} from 'react-test-renderer/shallow';

import CardFilter from './CardFilter';
import Store from "../../middleware/Store";

const renderer = createRenderer();

describe('<Home />', () => {
    it('should render and match the snapshot', () => {
        renderer.render(<Store><CardFilter/></Store>);
        const renderedOutput = renderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });
});
