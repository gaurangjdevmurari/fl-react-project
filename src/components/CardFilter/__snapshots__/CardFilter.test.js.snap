// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render and match the snapshot 1`] = `
<Context.Provider
  value={
    Array [
      Object {
        "format": "",
        "query": "",
      },
      [Function],
    ]
  }
>
  <CardFilter />
</Context.Provider>
`;
